src/tools/metadata-to-ipfs.ts

```diff
@@ -0,0 +1,41 @@
+import fs from "fs";
+// @ts-ignore
+import pinataSDK from "@pinata/sdk";
+import { NFTMetadata } from "../rmrk1.0.0/classes/nft";
+
+const pinata = pinataSDK(process.env.PINATA_KEY, process.env.PINATA_SECRET);
+
+const defaultOptions = {
+  pinataOptions: {
+    cidVersion: 1,
+  },
+};
+
+export const pinToIpfs = async (filePath: string, name?: string) => {
+  const options = { ...defaultOptions, pinataMetadata: { name } };
+  try {
+    const readableStreamForFile = fs.createReadStream(filePath);
```

Comment:
So this will crash if image is not readable or present, as stated above. Check for it please.

Reply:
Added validation